app/src/main/java/net/squanchy/tweets/service/TweetSpannedTextBuilder.kt

```diff
@@ -0,0 +1,95 @@
+package net.squanchy.tweets.service
+
+import android.text.Html
+import android.text.SpannableStringBuilder
+import android.text.Spanned
+import com.twitter.sdk.android.core.models.HashtagEntity
+import com.twitter.sdk.android.core.models.MentionEntity
+import com.twitter.sdk.android.core.models.UrlEntity
+import net.squanchy.tweets.view.TweetUrlSpan
+import net.squanchy.tweets.view.TweetUrlSpanFactory
+import java.util.regex.Pattern
+
+internal fun TweetUrlSpanFactory.applySpansToTweet(
```

Comment:
Not sure why this is an extension function instead of being in the class?

Reply:
Yeah good point, I wasn't really sure about this, at the end I just thought that this isn't really a member function but more of a pure function, you pass an input and you get an output, no state at all. At the same time I was bothered by having so many input parameters, so I guess in this case it was mostly about readability. I'm a bit concerned of having it public / internal, but I guess the typing here specific enough not to let room for abuse. But in case I can revert this to be a normal top level function

Reply:
Yeah I think in this case I'd rather have this in the class, if only for encapsulation... if you converted the whole thing to pure functions, that would be a different situation, but I wouldn't now anyway.

Reply:
Makes sense, will do